src/System.Console/src/System/ConsolePal.Windows.cs

```diff
@@ -1207,7 +1207,8 @@ private static unsafe int WriteFileNative(IntPtr hFile, byte[] bytes, int offset
                     {
                         int numBytesWritten;
                         writeSuccess = (0 != Interop.Kernel32.WriteFile(hFile, p + offset, count, out numBytesWritten, IntPtr.Zero));
-                        Debug.Assert(!writeSuccess || count == numBytesWritten);
+                        // In some cases we have seen numBytesWritten returned that is twice count;
+                        // so we aren't asserting the value of it.
```

Comment:
Link to the issue?  Otherwise, how does someone reading this later know what the "cases" were, whether the problem has been fixed, etc.?

Reply:
I avoid putting links to issues into the implementation code (in the test code, maybe). It makes it look messy, like a set of bug fixes. It's easy to do git blame and get the full PR. If you feel strongly, I can do it here?

Reply:
FWIW, I feel almost the exact opposite.  I find that leaving notes like this leaves a reader with more questions than it answers, even though the whole purpose of such a comment is to help a reader.  It's often not easy to find code as it moves around, especially across refactorings, taking a fair amount of effort sometimes to track changes over a prolonged period of time.  And even if it is feasible, having to take the time to go and do the research breaks me from my flow of looking at and understanding the code.  In my opinion, if it's worth a comment like this, it's also worth a TODO with an issue number, and then later a subsequent commit/PR later that either fixes it (e.g. puts back an assert or throw), clarifies it with necessary details (e.g. outlining the exact situations where it might return a different value), or removes it entirely.

I realize others can feel differently about it, and it's up to you how to handle this case.  I don't however want a general policy though that such TODOs with issue numbers shouldn't be included in src.

Reply:
Sure, OK, as long as we're selective about it.